### PR TITLE
docs: Fixup Docker quickstart instructions

### LINKED ISSUE #6860: Quickstart with Docker
https://vector.dev/docs/setup/quickstart/ assumes you will download and execute a binary on your host system which is a bit of an ask. I expected a `docker run timberio/vector:latest-alpine` style example 

### Use-cases

For interested users to better experiment with vector in a slightly sandboxed way.

### Attempted Solutions

This doesn't work. I am not sure why:

```
(ins)[hendry@t14s logs]$ cat vector.toml
[sources.in]
  type = "stdin"

[sinks.out]
  inputs = ["in"]
  type = "console"
  encoding.codec = "text"
(ins)[hendry@t14s logs]$ cat run.sh
echo 'Hello World!' | docker run -i \
  -v ~/vector.toml:/etc/vector/vector.toml:ro \
  -p 8383:8383 timberio/vector:latest-alpine
```

It just exits `INFO vector::shutdown: All sources have finished.`

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Closes #6860

### CODE REVIEW

File: website/content/en/docs/setup/quickstart.md [hunk 1]
```diff
@@ -79,7 +78,7 @@ The `inputs` option of the `sinks.out` component tells Vector where this sink's
 That's it for our first config. Now let's pipe an event through it:
 
 ```shell
-echo 'Hello world!' | vector --config ./vector.toml
```
Comment (neuronull) [inline website/content/en/docs/setup/quickstart.md:82]: Curious why the `--config` option not needed here?
I just ran and see it appears to be defaulting to /etc/vector/.

Comment (spencergilbert) [reply website/content/en/docs/setup/quickstart.md:82]: The config above ln60 is written locally to `./vector.toml` and then mounted to the `/etc/vector/vector.toml` path due to `-v $(pwd)/:/etc/vector/`

Comment (neuronull) [reply website/content/en/docs/setup/quickstart.md:82]: I see now. Thanks for clarifying!

Comment (spencergilbert) [reply website/content/en/docs/setup/quickstart.md:82]: I don't _super_ love it, but it's functional again. Happy to take other suggestions on cleaning things up.

**Pull request merged**